### PR TITLE
Tracks: Add seperate resourcelist class, docs, permissions for direct…

### LINKED ISSUE #4106: Add seperate resourcelist class, docs, permissions for direct endpoints of tracks
parent issue: #4101

### PR DESCRIPTION
**Pull request by shubham-padia**
fixes #4106

### CODE REVIEW

File: app/api/tracks.py [hunk 1]
```diff
@@ -64,13 +66,24 @@ def valid_color(self, data):
...
 
-class TrackList(ResourceList):
+class TrackListPost(ResourceList):
     """
     List and create Tracks
     """
     def before_post(self, args, kwargs, data):
         require_relationship(['event'], data)
+        if not has_access('is_coorganizer', event_id=data['event']):
+            raise ForbiddenException({'source': ''}, 'Co-organizer access is required.')
```
Comment (enigmaeth) [inline app/api/tracks.py]: @shubham-padia Shouldn't this be `track co-oragnizer` ? Not sure, correct me if I am wrong.

Comment (shubham-padia) [inline app/api/tracks.py]: Yep, will change it

File: app/api/__init__.py [hunk 2]
```diff
@@ -216,6 +216,7 @@
 api.route(SponsorRelationship, 'sponsor_event', '/sponsors/<int:id>/relationships/event')
 
 # tracks
+api.route(TrackListPost, 'track_list_post', '/tracks')
 api.route(TrackList, 'track_list', '/tracks', '/events/<int:event_id>/tracks',
```
Comment (enigmaeth) [inline app/api/__init__.py]: I think we don't need `tracks` endpoint here also.

File: docs/api/api_blueprint.apib [hunk 3]
```diff
@@ -6265,9 +6264,8 @@ Data related to the various tracks (e.g, python, AI, etc.) associated with the v
 | `color` | Color code of the track | string | **yes** |
 | `font-color` | Font Color of the track **(black/white: no user role - automatically updated)** | string | - |
 
-## Tracks Collection [/v1/events/{event_identifier}/tracks]
-+ Parameters
-    + event_identifier: 1 (string) - identifier or event id of the event. (b8324ae2 is an example of identifier)
+## Tracks Collection [/v1/tracks]
```
Comment (enigmaeth) [inline docs/api/api_blueprint.apib]: @shubham-padia  Add `page, size, sort, filter` here ??

File: app/api/__init__.py (hunk 2 above)
Comment (poush) [inline app/api/__init__.py]: @enigmaeth It is needed for GET method.

File: app/api/tracks.py (hunk 1 above)
Comment (shubham-padia) [inline app/api/tracks.py]: @enigmaeth Resolved

File: docs/api/api_blueprint.apib (hunk 3 above)
Comment (shubham-padia) [inline docs/api/api_blueprint.apib]: @enigmaeth Resolved

File: app/api/__init__.py (hunk 2 above)
Comment (enigmaeth) [inline app/api/__init__.py]: TrackListPost allows GET and POST both on `/tracks`, right and track_organiser access is checked in before_post so we can make GET request here also. Why do we need `/tracks` under TrackList also?

Comment (shubham-padia) [inline app/api/__init__.py]: @enigmaeth Yep, right. Resolved.

**Pull request merged**